script.js

```diff
@@ -1,4 +1,72 @@
+//win loss record
+var numOfWin = 0;
+var numOfLose = 0;
+var numOfDraw = 0;
+var winPercentage = 0;
+
+//User name
+var currentGameMode = "waiting for username";
+var userName = "";
+
+// Generate a random integer from 1 to max inclusive
+var generateRandomInteger = function (max) {
+  var randomDecimal = Math.random() * max;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+//Generate scissors, paper or stone
+var generateSPS = function () {
+  var numSPSInRepertoire = 3;
+  var random = generateRandomInteger(numSPSInRepertoire);
+  if (random == 1) {
+    return "scissors";
+  }
+  if (random == 2) {
+    return "paper";
+  }
+  if (random == 3) {
+    return "stone";
+  }
+};
+//the game
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "";
+  if (currentGameMode == "waiting for username") {
+    //set the name
+    userName = input;
+    currentGameMode = "ScissorsPaperStone";
+    return (myOutputValue =
+      "Welcome " +
+      userName +
+      "! Time for you to make a move! Scissors, Paper or Stone?");
+  } else if (currentGameMode == "ScissorsPaperStone") {
+    var winPercentage = 100 * (numOfWin / (numOfWin + numOfLose + numOfDraw));
+    var genericOutput = `${userName}'s play: ${input}. Program's play: ${randomSPS}.`;
+    var randomSPS = generateSPS();
+  }
```

Comment:
Good work on the return statements :) 

However it would be alot cleaner if this portion was used without the else if. 
You also don't need to put the variable myOutputValue  in when you return 

For example: 
```
  if (currentGameMode == "waiting for username") {
    //set the name
    userName = input;
    currentGameMode = "ScissorsPaperStone";
    myOutputValue = "Welcome " + userName + "! Time for you to make a move! Scissors, Paper or Stone?"
    return myOutputValue 
  } 
  
  if (currentGameMode == "ScissorsPaperStone") {
    var winPercentage = 100 * (numOfWin / (numOfWin + numOfLose + numOfDraw));
    var genericOutput = `${userName}'s play: ${input}. Program's play: ${randomSPS}.`;
    var randomSPS = generateSPS();
  }
```